torch/_functorch/deprecated.py

```diff
@@ -78,10 +78,11 @@ def jvp(func: Callable, primals: Any, tangents: Any, *, strict: bool = False, ha
     return _impl.jvp(func, primals, tangents, strict=strict, has_aux=has_aux)
 
 def jacrev(func: Callable, argnums: Union[int, Tuple[int]] = 0, *, has_aux=False,
-           chunk_size: Optional[int] = None,
+           chunk_size: Optional[int] = None, randomness: str = 'error',
```

Comment:
I'm confused:
- jacrev runs the forward pass without vmap, but runs the backward pass with vmap (depending on the chunk_size)
- there should be no randomness in backward passes, so we don't need the randomness argument.

How did we run into this?

Reply:
> How did we run into this?

Basically, compiling tests with dynamo on. Commit https://github.com/pytorch/pytorch/pull/116050/commits/321d022ac8b29e37021123aeba16e865855ef07d shows the tests I had to change to fix the CI failures.

Reply:
Debug this a bit more and the culprit of this are the `torch.randn`/`torch.randn_like` calls that happen before `jacrev`/`jacfwd` are executed. 

https://github.com/pytorch/pytorch/blob/56d7a47806ff918e8dfe5d137d857981c72b0523/test/functorch/test_ops.py#L1661-L1671

For instance, in the test above, if I change the `torch.randn_like` call by `cotangents = torch.empty_like(...)` the test passes.

For now, I guess I'll just close this PR and decorate all the related tests with `@xfailIfTorchDynamo` and get back to it once #116050 gets merged.

Reply:
This seems serious, because vmap is somehow being turned on when it is not being called! We should dig into this some more.

Reply:
I'm on it. Want to at least get the vmap fallback working properly before jump into another issue.

Reply:
I think I found the cause for this. Vmap internally calls `_vmap_increment_nesting` and during tracing if Dynamo fails for some reason, the counterpart `_vmap_decrement_nesting` is not called. Later on, any call to `torch.rand*` will be dispatched to its batched version.

Reply:
Oh I see. We should set up `_vmap_increment_nesting` as a context manager. Dynamo understand how to undo context managers when tracing

Reply:
See https://github.com/pytorch/pytorch/blob/0f0020d76f58294ddb23f34a7e33b89903ca126c/torch/_dynamo/variables/ctx_manager.py#L25 for examples